File: backend/src/main/java/com/lifescale/backend/profile/entity/UserProfile.java
package com.lifescale.backend.profile.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;

/**
 * 用户个人资料（展示型）：昵称 / 头像 / 问候语 / 每日提示。与 {@code ls_user} 1:1。
 * 仅存可由用户自行编辑的展示资料，账号与鉴权信息仍在 {@code ls_user}。
 */
@Entity
@Table(name = "ls_user_profile")
@EntityListeners(AuditingEntityListener.class)
public class UserProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false, unique = true)
    private Long userId;

    @Column(nullable = false, length = 64)
    private String nickname;

    @Column(name = "avatar_url", length = 512)
    private String avatarUrl;

    @Column(nullable = false, length = 100)
    private String greeting;

    @Column(name = "motivational_quote", nullable = false, length = 200)
    private String motivationalQuote;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    public UserProfile() {
    }

    /** 以默认资料构造某用户的资料行。 */
    public static UserProfile withDefaults(Long userId, String nickname) {
        UserProfile p = new UserProfile();
        p.userId = userId;
        p.nickname = nickname == null || nickname.isBlank() ? "用户" : nickname;
        p.avatarUrl = null;
        p.greeting = DEFAULT_GREETING;
        p.motivationalQuote = DEFAULT_MOTIVATIONAL_QUOTE;
        return p;
    }

    public static final String DEFAULT_GREETING = "早安";
    public static final String DEFAULT_MOTIVATIONAL_QUOTE = "专注当下，重视行动，让每一天都成为进步的刻度。";

    public Long getId() { return id; }
    public Long getUserId() { return userId; }
    public String getNickname() { return nickname; }
    public void setNickname(String nickname) { this.nickname = nickname; }
    public String getAvatarUrl() { return avatarUrl; }
    public void setAvatarUrl(String avatarUrl) { this.avatarUrl = avatarUrl; }
    public String getGreeting() { return greeting; }
    public void setGreeting(String greeting) { this.greeting = greeting; }
    public String getMotivationalQuote() { return motivationalQuote; }
    public void setMotivationalQuote(String motivationalQuote) { this.motivationalQuote = motivationalQuote; }
    public Instant getCreatedAt() { return createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
}

File: desktop/src/shared/types/userProfile.ts
/**
 * 用户个人资料（展示型），与后端 UserProfileDTO 对齐。
 * 字段与后端 /api/user/profile 一一对应。
 */
export interface UserProfile {
  /** 昵称（展示名）。 */
  nickname: string;
  /** 头像 URL，留空则前端按昵称首字渲染。 */
  avatarUrl: string;
  /** 问候语，如「早安」。 */
  greeting: string;
  /** 每日提示 / 励志金句。 */
  motivationalQuote: string;
}

/**
 * 个人资料更新载荷：部分更新，未提供（undefined）的字段不更新。
 */
export interface UpdateUserProfilePayload {
  nickname?: string;
  avatarUrl?: string;
  greeting?: string;
  motivationalQuote?: string;
}

/**
 * 未登录（本地模式）使用的固定默认资料。只读，不会变化。
 * greeting / motivationalQuote 在今日页等处直接展示，昵称留空（侧边栏显示「未登录」）。
 */
export const DEFAULT_USER_PROFILE: UserProfile = {
  nickname: '',
  avatarUrl: '',
  greeting: '早安',
  motivationalQuote: '专注当下，重视行动，让每一天都成为进步的刻度。',
};
